File: javascript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
	constructor(str) {
		this.arr=str.split('\n').map(s=>s.split(/\s/).map(Number));
	}

	get rows() {
		return this.arr
	}

	get columns() {
		return this.arr[0].map((_,index)=>this.arr.map(row=>row[index]))
	}
}

File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix_li = [i.split(' ') for i in matrix_string.split('\n')]

    def row(self, index):
        return list(map(int, self.matrix_li[index - 1]))

    def column(self, index):
        return [int(i[index - 1]) for i in self.matrix_li]
